source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -87,6 +87,17 @@ std::string InstanceImplPosix::fileReadToEnd(const std::string& path) {
 }
 
 bool InstanceImplPosix::illegalPath(const std::string& path) {
+  // Special case, allow /dev/fd/* access here so that config can be passed in a
+  // file descriptor from an execing bootstrap script. The reason we do this
+  // _before_ canonicalizing the path is that different unix flavors implement
+  // /dev/fd/* differently, for example on linux they are symlinks to /dev/pts/*
+  // which are symlinks to /proc/self/fds/. On BSD (and darwin) they are not
+  // symlinks at all. To avoid lots of platform, specifics, we whitelist
+  // /dev/fd/* _before_ resolving the canonical path.
+  if (absl::StartsWith(path, "/dev/fd/")) {
+    return false;
+  }
```

Comment:
This seems to be the simplest approach here rather than trying to research all the different OS implementations of this path and white list every possible ending point.

I'm not sure if there are any specific downsides of bypassing `realpath` here?

Reply:
Sorry for the delay, been catching up on backlog after OOO. I think this is fine for now, but can you leave a TODO to switch to `https://en.cppreference.com/w/cpp/filesystem/absolute` when we have C++17? The main downside of what you have is that there is no relative path normalization, which we want independent of symlink following.
/wait

Reply:
Added the TODO. I hope this is appropriate - I added it under your username as it seems to be more likely you will have context for this later than I will!